### PR TITLE
Refactor custom network forms

### PR DESCRIPTION
**Pull request by chidg**
Fixes validation race condition, also refactors substrate form to make it more legible

### CODE REVIEW

**Review by alecdwm — DISMISSED** [review]
LGTM!!!

File: apps/extension/src/core/domains/chains/types.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ export type RequestUpsertCustomChain = {
   nativeTokenLogoUrl: string | null
   accountFormat: string | null
   subscanUrl: string | null
-  rpcs: { url: string }[]
+  rpcs: { url: string; genesisHash?: SignerPayloadGenesisHash }[]
```
Comment (0xKheops) [inline apps/extension/src/core/domains/chains/types.ts]: not sure if we should do this, this defines the argument for backend call which doesn't need this property

Comment (alecdwm) [reply apps/extension/src/core/domains/chains/types.ts]: Yeah this stood out to me too, but I couldn't think of a better place to put the genesisHash instead

Comment (chidg) [reply apps/extension/src/core/domains/chains/types.ts]: ok, in that case I think we should separate data needed for the form itself from data needed in the backend and run it through a transform. Sound ok?

Comment (0xKheops) [reply apps/extension/src/core/domains/chains/types.ts]: sounds great

**Review by 0xKheops — APPROVED** [review]
LGTM !

**Pull request merged**